pkg/network/client_connection.go

```diff
@@ -119,12 +122,9 @@ func (t *clientTcpConnection) startPolling(initialConn net.Conn) {
 		}
 	}(initialConn)
 	go func() {
-		<-t.ctx.Done()
+		<-stoppedConsumingCtx.Done()
```

Comment:
Likewise as below, this goroutine (and the stoppedConsumingCtx) can go if you move the cleanup to the `case <-t.ctx.Done():` above.